impl/httpengine.go

```diff
@@ -107,7 +107,7 @@ func (c ChukonuHttpResponse) Dump() ([]byte, error) {
 	return httputil.DumpResponse(c.Response, false)
 }
 
-func NewHttpEngine(config core.ChukonuConfig) *HttpEngine {
+func NewHttpEngine(config core.ChukonuConfig) core.Engine {
```

Comment:
Because in the test plan, we want the users to be able to specify the engine to use. Then in the pool.go and lifecycle.go, we don't need to hardcode any of the detailed impl classes, we can just get the detailed impl from the test plan.